spock-specs/src/test/groovy/org/spockframework/smoke/ast/AstSpec.groovy

```diff
@@ -387,4 +387,158 @@ class TestSpec extends Specification {
     then:
     textSnapshotter.assertThat(result.source).matchesSnapshot()
   }
+
```

Comment:
IMO, these should be grouped in their own spec. We already have a package for `org.spockframework.smoke.ast.condition`.